app/models/vote.rb

```diff
@@ -0,0 +1,4 @@
+class Vote < ApplicationRecord
+  belongs_to :ice_cream, counter_cache: true
```

Comment:
You should also have a validation that the combo of user_id and ice_cream_id  is unique to keep someone from voting on it twice.

```suggestion
  belongs_to :ice_cream, counter_cache: true
  validates :ice_cream_id, uniqueness: {scope: :user_id, 
    message: "You cannot vote for the same icecream twice" }
```